### PR TITLE
[State Observability] Support output formatting

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR supports various output formatting. By default, we support `yaml` format. But this can be changed depending on the UX research we will conduct in the future. 

```
1cb0f4b51a5799e0360a66db01000000:
  actor_id: 1cb0f4b51a5799e0360a66db01000000
  class_name: A
  state: ALIVE
f90ba34fa27f79a808b4b5aa01000000:
  actor_id: f90ba34fa27f79a808b4b5aa01000000
  class_name: A
  state: ALIVE
```

Table format is not supported yet. We will support this once we enhance the API output (which I will create an initial API review soon). 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — CHANGES_REQUESTED** [review]

File: python/ray/experimental/state/state_cli.py [hunk 1]
```diff
@@ -20,6 +23,34 @@
...
+    elif format == "json":
+        print(json.dumps(state_data))
+    elif format == "table":
+        raise NotImplementedError("Table formatter is not implemented yet.")
+    else:
+        raise ValueError(
+            f"Unexpected format: {format}. "
+            "Supported formatting: [default | json | table]"
+        )
+
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply python/ray/experimental/state/state_cli.py:52]: sgtm! The enum is only for internal usage right? (The API will require to specify the raw string?)

File: python/ray/experimental/state/state_cli.py [hunk 2]
```diff
@@ -20,6 +23,34 @@
...
 
+def format_print(data: dict, indentation: int = 1):
+    for k, v in data.items():
+        tabs = "".join(["\t" for _ in range(indentation)])
+        print(f"{tabs}{k}:", end="")
+        if isinstance(v, dict):
+            print()
+            format_print(v, indentation=indentation + 1)
+        else:
+            print(f" {v}")
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply python/ray/experimental/state/state_cli.py:34]: Actually I will delete it and use yaml as a default format instead.

File: python/ray/experimental/state/state_cli.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/experimental/state/state_cli.py]: You can just do `str(enum_value)` if you have the `Enum` inherit from str

File: python/ray/experimental/state/state_cli.py [hunk 3]
```diff
@@ -19,6 +22,29 @@
     list_runtime_envs,
 )
 
+AVAILABLE_FORMAT = ["default", "yaml", "json", "table"]
```
Comment (edoakes) [inline python/ray/experimental/state/state_cli.py:25]: still prefer enum instead for readability/type checking but unblocking merge, your call...

**Review by edoakes — APPROVED** [review]

File: python/ray/experimental/state/state_cli.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/experimental/state/state_cli.py]: and conversely you can cast the str to enum like: `MyEnum(str_val)`

File: python/ray/experimental/state/state_cli.py (hunk 3 above)
Comment (rkooo567) [reply python/ray/experimental/state/state_cli.py]: I used enum! Please take a look. I will merge the PR around tomorrow!

**Comment by rkooo567** [comment]
Failed tests seem unrelated

**Pull request merged**